src/material/menu/menu-trigger.ts

```diff
@@ -121,6 +121,10 @@ export class MatMenuTrigger implements AfterContentInit, OnDestroy {
     this._menuCloseSubscription.unsubscribe();
 
     if (menu) {
+      if (menu === this._parentMenu) {
```

Comment:
Let's only do this in dev mode
(thinking long term to when we'll be able to drop dev mode checks from the prod output)